cmd/commands/application/command.go

```diff
@@ -0,0 +1,137 @@
+/*
+ * Copyright (C) 2017 The "MysteriumNetwork/node" Authors.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package application
+
+import (
+	"fmt"
+	"path/filepath"
+
+	"github.com/mysterium/node/cmd"
+	command_cli "github.com/mysterium/node/cmd/commands/cli"
+	"github.com/mysterium/node/cmd/commands/license"
+	"github.com/mysterium/node/cmd/commands/run"
+	"github.com/mysterium/node/cmd/commands/version"
+	"github.com/mysterium/node/core/node"
+	"github.com/mysterium/node/metadata"
+	tequilapi_client "github.com/mysterium/node/tequilapi/client"
+	"github.com/mysterium/node/utils"
+	"github.com/urfave/cli"
+)
+
+// NewCommand function creates application master command
+func NewCommand() (*cli.App, error) {
+	cli.VersionPrinter = func(ctx *cli.Context) {
+		versionCommand.Run(ctx)
+	}
+
+	app := cli.NewApp()
+	app.Usage = "VPN server and client for Mysterium Network https://mysterium.network/"
+	app.Authors = []cli.Author{
+		{`The "MysteriumNetwork/node" Authors`, "mysterium-dev@mysterium.network"},
+	}
+	app.Version = metadata.VersionAsString()
+	app.Copyright = licenseCopyright
+
+	if err := cmd.RegisterDirectoryFlags(&app.Flags, &options); err != nil {
+		return app, err
```

Comment:
Is it intentional to return app object and not nil error? Normal go convensions say that in case of error - returned object contains undefined state and should not be used (from caller perspective)

Reply:
fixed to `return nil, err`